### PR TITLE
Kernel shutdown when closing notebook tab

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Issue: https://github.com/jupyterlab/jupyterlab/issues/5241
This is a simplified approach of PR https://github.com/jupyterlab/jupyterlab/pull/5914 as suggested by @ellisonbg to target v1.0.

## Code changes

To shutdown a kernel when closing a tab, a call `session.shutdown` is performed when `ClientSession` is disposed. This is not done in `NotebookPanel.dispose` to prevent a kernel shutdown when multi-views on a notebook are opened.

To pass the parameter from the settings to the `ClientSession`, a new attribute has been added in `kernelPreference` provided by the widget factory.

## User-facing changes

New settings:
*Notebook -> Shutdown kernel*

## Note

If activated, closing the notebook panel with an associated console view will result in the kernel shutdown. Therefore the opened console will lost its kernel connection.

### CODE REVIEW

**Comment by jasongrout** [comment]
> If activated, closing the notebook panel with an associated console view will result in the kernel shutdown. Therefore the opened console will lost its kernel connection.

Does this also mean that if I open a console associated with a notebook, then close the console, the notebook's kernel will shut down?

**Comment by fcollonval** [comment]
> Does this also mean that if I open a console associated with a notebook, then close the console, the notebook's kernel will shut down?

Nothing has been implemented in this PR regarding the console. In your scenario, the notebook kernel will continue to run.

I try to implement a similar setting for the console. But I hit your point as I did not figure out how could I distinguish console created from scratch, from file or from notebook. I thought I could test whether the path is provided or not at console creation. But this will result in a kernel still running when closing console linked to a python script for example. I don't think that this will be expected by the user.

I propose to limit this PR to notebook tab.

File: packages/notebook-extension/schema/tracker.json [hunk 1]
```diff
@@ -284,6 +284,12 @@
         "codeFolding": false
       }
     },
+    "kernelShutdown": {
+      "title": "Shutdown kernel",
+      "description": "Whether to shutdown or not the kernel when closing a notebook.",
```
Comment (jasongrout) [inline packages/notebook-extension/schema/tracker.json]: "shut down" - http://notaverb.com/shutdown

File: packages/notebook-extension/schema/tracker.json [hunk 2]
```diff
@@ -284,6 +284,12 @@
         "codeFolding": false
       }
     },
+    "kernelShutdown": {
+      "title": "Shutdown kernel",
```
Comment (jasongrout) [inline packages/notebook-extension/schema/tracker.json]: "Shut down"

**Comment by fcollonval** [comment]
Thanks for the review @jasongrout 

This is good to go for me.

**Pull request merged**